enterprise/api/v1/views.py

```diff
@@ -248,14 +297,73 @@ class EnterpriseCustomerEntitlementViewSet(EnterpriseReadOnlyModelViewSet):
     ordering_fields = FIELDS
 
 
-class EnterpriseCatalogViewSet(viewsets.ViewSet):
+class EnterpriseCustomerCatalogApiViewSet(EnterpriseWrapperApiViewSet):
     """
-    API views for `enterprise customer catalogs` api endpoint.
+    API Views for performing search through course discovery at the ``enterprise-catalogs`` API endpoint.
     """
-    serializer_class = serializers.EnterpriseCourseCatalogReadOnlySerializer
-    permission_classes = (permissions.IsAuthenticated,)
-    authentication_classes = (JwtAuthentication, BearerAuthentication, SessionAuthentication,)
-    throttle_classes = (ServiceUserThrottle,)
+
+    permission_classes = (permissions.IsAuthenticated, IsStaffUserOrLinkedToCommonEnterprise,)
+    serializer_class = serializers.EnterpriseCustomerCatalogApiPaginatedResponseReadOnlySerializer
+    cross_check_model = models.EnterpriseCustomerCatalog
+
+    def retrieve(self, request, pk=None):  # pylint: disable=invalid-name
+        """
+        DRF view to list all content from course discovery that matches the EnterpriseCustomerCatalog's query.
+
+        Arguments:
+            request (HttpRequest): Current request.
+            pk (string): Identifier for the EnterpriseCustomerCatalog instance that contains the necessary query.
+
+        Returns:
+            (Response): DRF response object containing a paginated result of searching the course discovery service.
+        """
+        if 'programs' in request.path:
+            # For the Programs endpoint, just search course discovery for the UUID.
```

Comment:
Ah this comment made more sense when I had `uuid` as `pk`, but had to change it to be *actually* `pk` because that's what the function apparently expects. I'll change it now.